app/src/main/java/com/sdp/movemeet/utility/DistanceCalculator.java

```diff
@@ -47,18 +61,28 @@ public void setActivities(ArrayList<Activity> activities) {
         }
     }
 
-
+    /**
+     * Called in MainMapFragment to return the list of N closest activities.
+     * @param n the number of activities to return
+     */
     @RequiresApi(api = Build.VERSION_CODES.N)
     public ArrayList<Activity> getTopActivities(Integer n) {
         if (sorted) {
-            Stream<Activity> topActivitiesStream = activityDistanceMap.subList(0, n).stream().map(Pair::getKey);
```

Comment:
why do you pass it in a loop ?

Reply:
`activityDistanceMap` is an `ArrayList` of `Pair`s (it's kind of like a `HashMap`).

When it's sorted, it looks like:
`{(ActivityA, 0.1), (ActivityB, 0.2), (ActivityC, 0.3)}`

So we want to return the first element ("the key") of every `Pair` in the `ArrayList`, that's why we call `.map(Pair::getKey)`.

Hope this helps!